thumbor/filters/watermark.py

```diff
@@ -207,3 +210,20 @@ async def watermark(self, url, x, y, alpha, w_ratio=False, h_ratio=False):
                 raise error
             logger.warning("bad watermark")
             raise tornado.web.HTTPError(500)
+
+    def validate(self, url):
+        if not hasattr(self.context.modules.loader, "validate"):
+            return True
+
+        is_valid = self.context.modules.loader.validate(self.context, url)
+
+        if not is_valid:
+            logger.warning(
+                (
+                    'Request denied because the specified url "%s" was '
+                    "not identified by the loader as a valid url"
+                ),
+                url,
+            )
+
+        return is_valid
```

Comment:
* Nit 1: The warning message is long, especially when compared to other logged messages. Also, a request might not make sense in the context of a filter. Something like “watermark source not allowed” would make more sense, IMO.

* Nit 2: The extra variable (and its added cognitive load) makes it a bit slower (and less easy to read) than returning a boolean instead, e.g.:

  ```suggestion
          if not self.context.modules.loader.validate(self.context, url):
              logger.warning('watermark source not allowed: "%s"', url)
              return False
  
          return True
  ```
  
* Nit 3: I know, this code is duplicated from the following source:

  https://github.com/thumbor/thumbor/blob/1784b200107225581379f06eb688bbe681d6fe64/thumbor/handlers/__init__.py#L746-L761

  And it would be desirable to have the URL validation functionality in a single place.

So, I'd say we merge this as is, and solve the aforementioned nits in a future pull request.

Reply:
Thanks for your insights.
I implemented the suggestions and commited again. 
Could you please re-review?